test/e2e/specs/site-editor/template-part.spec.js

```diff
@@ -309,4 +309,70 @@ test.describe( 'Template Part', () => {
 
 		await expect( paragraph ).toBeVisible();
 	} );
+
+	test( 'can import a widget area into an empty template part', async ( {
+		admin,
+		siteEditor,
+		editor,
+		page,
+	} ) => {
+		await admin.visitSiteEditor();
+		await siteEditor.enterEditMode();
+
+		// Add a block and select it.
+		await editor.insertBlock( {
+			name: 'core/template-part',
+		} );
+
+		// Open Block Inspector -> Advanced.
+		await editor.openDocumentSettingsSidebar();
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+
+		// Import the widget area.
+		await page
+			.getByRole( 'combobox', { name: 'Import widget area' } )
+			.selectOption( 'sidebar-1' );
+		await page.getByRole( 'button', { name: 'Import' } ).click();
+
+		// Verify that the widget area was imported.
+		await expect(
+			editor.canvas.locator( '[data-type="core/search"]' )
```

Comment:
Nit: we can also do `editor.getBlocks()` and assert the array structure. See this [guide](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md) for example.

Reply:
Thanks! I was actually looking for that helper but couldn't find it.

In any case it doesn't help us here because the blocks we care about are within the template part entity and so not returned by `select( 'core/block-editor' ).getBlocks()`.